chalice/package.py

```diff
@@ -1,217 +1,285 @@
         }
-
-    def _generate_function_events(self, app, function_type):
-        # type: (Chalice, str) -> Dict[str, Any]
-        return getattr(
-            self, '_generate_' + function_type + '_function_events')(app)
-
-    def _generate_api_function_events(self, app):
-        # type: (Chalice) -> Dict[str, Any]
-        events = {}
-        for methods in app.routes.values():
-            for http_method, view in methods.items():
-                key_name = to_cfn_resource_name(
-                    view.view_name + http_method.lower())
-                events[key_name] = {
-                    'Type': 'Api',
-                    'Properties': {
-                        'Path': view.uri_pattern,
-                        'RestApiId': {'Ref': 'RestAPI'},
-                        'Method': http_method.lower(),
-                    }
-                }
-        return events
-
-    def _generate_authorizer_function_events(self, app):
-        # type: (Chalice) -> Dict[str, Any]
-        return {}
-
-    def _generate_rest_api(self, app, api_gateway_stage):
-        # type: (Chalice, str) -> Dict[str, Any]
-        swagger_definition = self._swagger_generator.generate_swagger(app)
-        properties = {
-            'StageName': api_gateway_stage,
-            'DefinitionBody': swagger_definition,
+        outputs['APIHandlerArn'] = {
+            'Value': {'Fn::GetAtt': ['APIHandler', 'Arn']}
         }
-        return {
-            'Type': 'AWS::Serverless::Api',
-            'Properties': properties,
+        outputs['EndpointURL'] = {
+            'Value': {
+                'Fn::Sub': (
+                    'https://${RestAPI}.execute-api.${AWS::Region}'
+                    # The api_gateway_stage is filled in when
+                    # the template is built.
+                    '.amazonaws.com/%s/'
+                ) % stage_name
+            }
         }
 
-    def _generate_iam_policy(self, config):
-        # type: (Config) -> Dict[str, Any]
-        return self._policy_generator.generate_policy_from_app_source(config)
+    # The various IAM roles/policies are handled in the
+    # Lambda function generation.  We're creating these
+    # noop methods to indicate we've accounted for these
+    # resources.
+
+    def _generate_managediamrole(self, resource, template):
+        # type: (models.ManagedIAMRole, Dict[str, Any]) -> None
+        pass
+
+    def _generate_filebasediampolicy(self, resource, template):
+        # type: (models.FileBasedIAMPolicy, Dict[str, Any]) -> None
+        pass
+
+    def _generate_autogeniampolicy(self, resource, template):
+        # type: (models.AutoGenIAMPolicy, Dict[str, Any]) -> None
+        pass
+
+    def _generate_deploymentpackage(self, resource, template):
+        # type: (models.DeploymentPackage, Dict[str, Any]) -> None
+        pass
+
+    def _default(self, resource, template):
+        # type: (models.Model, Dict[str, Any]) -> None
+        raise NotImplementedError(resource)
```

Comment:
I wonder if it makes sense to add a test for this? It looks like it would work but may be nice to double check.